### PR TITLE
Fixes bug in which Collect[] couldn't take inputs not from the subdag

### PR DESCRIPTION
**Pull request by elijahbenizzy**
This was due to an untested codepath that was getting the wrong key from upstream.

[Short description explaining the high-level reason for the pull request]

## Checklist

- [ ] PR has an informative and human-readable title (this will be pulled into the release notes)
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future TODOs are captured in comments
- [ ] Project documentation has been updated if adding/changing functionality.

### CODE REVIEW

File: tests/execution/test_executors.py [hunk 1]
```diff
@@ -230,24 +229,20 @@ def test_get_execotor_for_tasks_default_execution_manager(purpose, check):
...
+def test_end_to_end_parallelizable_with_overrides():
     dr = (
         driver.Builder()
-        .with_modules(tmp_module)
+        .with_modules(inputs_in_collect)
         .enable_dynamic_execution(allow_experimental_mode=True)
         .with_remote_executor(SynchronousLocalTaskExecutor())
     ).build()
-    assert dr.execute(["collect"])["collect"] == 46
+    assert dr.execute(["collect"], overrides={"par": [1, 2, 3]})["collect"] == 7
```
Comment (skrawcz) [inline tests/execution/test_executors.py:248]: this doesn't test overriding collect though?

Comment (skrawcz) [reply tests/execution/test_executors.py:248]: can you override identity?

Comment (elijahbenizzy) [reply tests/execution/test_executors.py:248]: Nope, you cannot override `identity` -- its in the middle of the block. Can test overriding `collect` as well.

Comment (elijahbenizzy) [reply tests/execution/test_executors.py:248]: Yes, but I don't have the time right now to dig in on that -- this is a patch and why its still in experimental. Will create an issue.

Comment (elijahbenizzy) [reply tests/execution/test_executors.py:248]: https://github.com/DAGWorks-Inc/hamilton/issues/301

**Review by skrawcz — APPROVED** [review]

**Pull request merged**